src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -262,13 +261,7 @@ def setup(self, trainer: "pl.Trainer") -> None:
         assert self.lightning_module is not None
         self.lightning_module._device = self.root_device
 
-        if is_overridden("configure_sharded_model", self.lightning_module):
```

Comment:
Now that there are uses for `configure_sharded_model` that aren't wrapping, we shouldn't skip wrapping if the hook is defined.